### PR TITLE
Add per-sidecar config dump to ADS debug

### PR DESCRIPTION
**Pull request by liamawhite**
This is a follow on from @rshriram's original PR which can be found here https://github.com/istio/istio/pull/4803

### CODE REVIEW

File: pilot/pkg/proxy/envoy/v2/ads.go [hunk 1]
```diff
@@ -365,6 +375,17 @@ func (s *DiscoveryServer) removeCon(conID string, con *XdsConnection) {
 		log.Errorf("ADS: Removing connection for non-existing node %v.", s)
 	}
 	delete(adsClients, conID)
+	if _, ok := adsSidecarIDConnectionsMap[con.modelNode.ID]; ok {
```
Comment (liamawhite) [inline pilot/pkg/proxy/envoy/v2/ads.go]: Don't particularly like this, should the slice instead be a `map[conID]*XdsConnection`?

Comment (rshriram) [reply pilot/pkg/proxy/envoy/v2/ads.go]: That was the edge case I told you about

**Comment by liamawhite** [comment]
Not sure what's going on here... The test passes locally.

```
Running tool: /usr/local/bin/go test -timeout 30s istio.io/istio/pilot/pkg/proxy/envoy/v2 -run ^TestAdsReconnectWithNonce$

ok  	istio.io/istio/pilot/pkg/proxy/envoy/v2	1.082s
Success: Tests passed.
```

File: pilot/pkg/proxy/envoy/v2/debug.go [hunk 2]
```diff
@@ -48,8 +49,6 @@ func (s *DiscoveryServer) InitDebug(mux *http.ServeMux, sctl *aggregate.Controll
 	})
 
 	mux.HandleFunc("/debug/edsz", edsz)
-
-	mux.HandleFunc("/debug/ldsz", ldsz)
```
Comment (costinm) [inline pilot/pkg/proxy/envoy/v2/debug.go]: Why remove ?

File: pilot/pkg/proxy/envoy/v2/debug.go [hunk 3]
```diff
@@ -375,22 +374,41 @@ func adsz(w http.ResponseWriter, req *http.Request) {
 		fmt.Fprintf(w, "Pushed to %d servers", len(adsClients))
 		return
 	}
-	adsClientsMutex.RLock()
-	data, err := json.Marshal(adsClients)
-	adsClientsMutex.RUnlock()
 
-	if err != nil {
-		_, _ = w.Write([]byte(err.Error()))
+	if proxyID := req.Form.Get("proxyID"); proxyID != "" {
```
Comment (costinm) [inline pilot/pkg/proxy/envoy/v2/debug.go]: 'id' ? 

Also can you add another option to list the IDs ( or maybe a brief "podID podIP" ) ?

File: pilot/pkg/proxy/envoy/v2/debug.go (hunk 2 above)
Comment (liamawhite) [reply pilot/pkg/proxy/envoy/v2/debug.go]: Listeners are outputted as part of the adsz, this is inline with @rshriram's initial PR.

File: pilot/pkg/proxy/envoy/v2/debug.go (hunk 3 above)
Comment (liamawhite) [reply pilot/pkg/proxy/envoy/v2/debug.go]: I'm not sure I understand what you're asking for?
